docs/advanced.md

```diff
@@ -0,0 +1,57 @@
+# Advanced
```

Comment:
I began work on some more documentation of these variant types to make them easier to understand and discover. Let me know what you think!